File: api/sources/feedbin.js
const {runQuery} = require('../db');
const {lookupCounter, lookupLength} = require('../metrics');

async function getLikes({sort, limit, offset}) {
  const query = `
    SELECT title, description, link, author, date_updated FROM likes
    ORDER BY date_updated ${sort}
    LIMIT ${limit} OFFSET ${offset};
  `;
  const result = await runQuery(query);
  lookupCounter.labels('likes').inc();
  lookupLength.labels('likes').set(result.length);
  return result;
}

module.exports = {getLikes};

File: api/feedbin.go
// SELECT title, description, link, author, date_updated FROM likes
// ORDER BY date_updated ${sort}
// LIMIT ${limit} OFFSET ${offset};

package main

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type FeedbinArticle struct {
	Title       string `json:"name"`
	Description string `json:"description"`
	Link        string `json:"link"`
	Author      string `json:"author"`
	DateUpdated string `json:"date_updated"`
}

func GetFeedbinHandler(w http.ResponseWriter, r *http.Request) {
	rows, err := db.Query(`
		SELECT title, description, link, author, date_updated FROM likes
		ORDER BY date_updated DESC
	`)
	articles := []FeedbinArticle{}
	defer rows.Close()
	for rows.Next() {
		article := FeedbinArticle{}
		err = rows.Scan(&article.Title, &article.Description, &article.Link, &article.Author, &article.DateUpdated)
		if err != nil {
			panic(err)
		}
		articles = append(articles, article)
	}
	b, _ := json.Marshal(articles)
	s := string(b)
	fmt.Fprintf(w, s)
}
